blackbox/docs/general/builtins/window-functions.rst

```diff
@@ -0,0 +1,69 @@
+.. highlight:: psql
+.. _window-functions:
+
+================
+Window Functions
+================
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Introduction
+============
+
+:ref:`Aggregation <aggregation>` functions can be executed over a subset of the
+rows returned by a query.
+Multiple rows might be connected or semantically related to the current row, in
+which case the business requirements might entail executing the aggregations
+on all related rows and emitting the result for the current row, as opposed to
+executing aggregations on the entire query result set and emitting the result
+just once (as it would happen by executing a plain 
+:ref:`aggregation function <aggregation>`).
+The set of related rows are defining the ``window`` the function will be
+executed against.
+
+Window functions enable an elegant way to express queries that include time
+series, moving averages or cumulative sums. Without window functions the user
+would either have to write some possibly very expensive correlated subqueries
+or implement them at the application level.
+
+Window definition
+=================
+
+In order to use the :ref:`aggregation functions <aggregation>` as 
```

Comment:
I'd also rephrase this a bit to make the aggregation functions less prominent. I think conceptually this shouldn't be about using aggregations as window functions, but rather be about using window functions. Aggregation *can* be used as window functions as well. But we'll have dedicated window functions later on.